### PR TITLE
Making the GitLab repoUrl picker description clearer

### PR DESCRIPTION
**Pull request by NishkarshRaj**
## Hey, I just made a Pull Request!

A minor update to the description but hopeful it would make life easy for the adopters.

For us at WH, we've been requested this for a while now. The field states `Owner` so people tend to fill the team name but it should be GitLab namespace.

Calling a Spade a Spade would make it easy to fill the form correctly.

Reference: https://docs.gitlab.com/ee/user/namespace/

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by NishkarshRaj** [comment]
For the maintainers, outside the scope of this PR but a brain dump:

- We should add the ability to override the default description on a configuration level.
- Invoke an API to the Git VCS hosting (not restricted to GitLab) to get a list of namespaces the signed-in user is allowed to create a repository in and showcase the list in a form of a dropdown.

**Review by jhaals — APPROVED** [review]
Thanks for improving the user experience 👍 

The scaffolder does support acting with the users credentials when creating/pushing code, however I'm not sure if that's integrated with the picker that you use when scaffolding. I'll have to defer to @benjdlambert. Feel free to create a separate issue or RFC with suggestions for things that can improve :)

**Pull request merged**
